rules/network/command_and_control_fin7_c2_behavior.toml

```diff
@@ -0,0 +1,55 @@
+[metadata]
+creation_date = "2020/07/06"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/07/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects a known command and control pattern in network events. The FIN7 threat group is known to use this
+command and control technique, while maintaining persistence in their target's network.
+"""
+false_positives = [
+    """
+    This rule could identify benign domains that are formatted similarly to FIN7's command and control algorithm. Alerts
+    should be investigated by an analyst to assess the validity of the individual observations.
+    """,
+]
+index = ["packetbeat-*"]
+language = "lucene"
+license = "Elastic License"
+name = "Possible FIN7 DGA Command and Control Behavior"
+note = "This activity has been observed in FIN7 campaigns."
+references = [
+    "https://www.fireeye.com/blog/threat-research/2018/08/fin7-pursuing-an-enigmatic-and-evasive-global-criminal-operation.html",
+]
+risk_score = 73
+rule_id = "4a4e23cf-78a2-449c-bac3-701924c269d3"
+severity = "high"
+tags = ["Elastic", "Network"]
+type = "query"
+
+query = '''
+event.category:(network OR network_traffic) AND type:(tls OR http) AND network.transport:tcp
+AND destination.domain:/[a-zA-Z]{4,5}\.(pw|us|club|info|site|top)/ AND NOT destination.domain:zoom.us
```

Comment:
any opportunities to tighten this down to more specific fin7 activity to avoid FPs like `zoom.us`?

Reply:
When I looked at a large dataset, the only FPs were from the `us` TLD and it was Zoom.

We could eliminate that TLD, but we could miss some C2 activity.